### DISCUSSION TITLE
Codespaces shutting down even when terminal output is happening

### DISCUSSION THREAD

**Discussion by EarthmanWeb** [community/community#60407] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

Hi there, I'm running codespaces in browser / Chrome app. Loving it so far, except this important thing:

I'm running into an issue where I can't run a persistent bash script in my codespace, because the codespace shuts down after the set time period (I upped it to 4 hours) regardless of terminal output.

I've seen it also shutdown when using the terminal just minutes before, when I have it installed on my Mac or iPad, in a Google Chrome App wrapper.

What am I missing here? Is this a documented bug? How can I be sure that the timeout period is being reset with each terminal output? Is there a way to debug this issue?

I see a related feature request here: https://github.com/orgs/community/discussions/35428
so it seems i am not the only one...

Expected behaviour - from:
https://docs.github.com/en/codespaces/getting-started/deep-dive#closing-or-stopping-your-codespace

> File changes from the editor and terminal output are counted as activity, so your codespace will not time out if terminal output is continuing.

**Comment by osortega** [comment]
Hello @EarthmanWeb!

Sorry for the friction, do you mind sharing a bit more about the process you are running and the output it's producing?
Codespaces is designed to maintain an interactive session with the user and will maintain the connection while users are interacting with the codespace.

If you leave something running for hours and close your laptop, the laptop goes to sleep or even if the tab is forgotten and ends up being suspended, the connection will eventually drop. Even if the codespace is still executing a process, the client will not detect activity on the user side and will eventually become idle and stop.

Is this something that could be happening in your scenario?

Thanks,
Osvaldo

**Comment by luketegel** [comment]
Have you tried keeping the Codespaces tab/window open in the background? I’ve had similar issues and this worked for me. Additionally are you running low on quota in your billing tab? (that could be an issue however I’m not sure).

Hope this helps!

**Comment by EarthmanWeb** [comment, marked as answer]
This appears to have been resolved.  

I am running codespaces in a Chrome App container.  I had the sleep disabled already.  No quota issues either.

I set the Chrome Memory Saver to be disabled for my codespace sites, in Chrome under: "3 dot menu > Settings > Performance > Memory Saver > Always keep these sites active" - that seems to have solved it!

**Comment by benonymus** [comment]
I am running into a similar issue. 
I am trying to keep a codespace running if there is external traffic to it.
I close the vscode view, and interact with the running application.
After the timeout period it is always killed even if it is being interacted with.
I tried to run the traffic logs in the terminal - the logs happen if there is traffic - but it is still killed.
I am trying to have this automated as much as possible so people can create a codespace and close the codespace itself and test on the running exposed application to do QA checks.
I am starting the app in the postStartCommand and I have a vscode task to open a terminal with the logs.
Any ideas?
